src/utils/errorHandler.ts

```
@@ -0,0 +1,37 @@
+        const result = await operation();
+        resolve(result);
+        return;
+      } catch (error) {
+        if (attempt === retries) {
+          reject(error);
+          return;
+        }
+        await new Promise(resolve => setTimeout(resolve, delay * attempt));
+      }
+    }
+  });
```

Biome (1.9.4)

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)